File: backend/app/models/invite.py
"""Team invite Pydantic models."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, EmailStr, Field

from app.models.enums import Role


class InviteCreate(BaseModel):
    """Payload for creating a team-scoped invite."""

    email: EmailStr
    role: Role = Field(
        default=Role.MEMBER,
        description="Role assigned when the invite is accepted.",
    )


class InviteResponse(BaseModel):
    """Invite record returned after creation or lookup."""

    id: UUID
    team_id: UUID
    email: EmailStr
    role: Role
    token: str
    invited_by: UUID
    expires_at: datetime
    accepted_at: datetime | None = None
    created_at: datetime


class InviteAcceptRequest(BaseModel):
    """Optional body when accepting an invite (user may already be logged in)."""

    full_name: str | None = Field(default=None, max_length=120)


class InviteAcceptResponse(BaseModel):
    """Result of a successful invite acceptance."""

    team_id: UUID
    role: Role
    membership_id: UUID

File: frontend/src/types/invite.ts
/** Invite types matching backend app.models.invite */

import type { Role } from "./roles";

export interface InviteCreate {
  email: string;
  role?: Role;
}

export interface Invite {
  id: string;
  team_id: string;
  email: string;
  role: Role;
  token: string;
  invited_by: string;
  expires_at: string;
  accepted_at: string | null;
  created_at: string;
}

export interface InviteAcceptResponse {
  team_id: string;
  role: Role;
  membership_id: string;
}
